scripts/tasks/utils.ts

```diff
@@ -46,26 +44,34 @@ export function getTsPathAliasesApiExtractorConfig(options: {
   tsConfigPath: string;
   packageJson: PackageJson;
 }) {
-  const rootOffset = offsetFromRoot(path.dirname(options.tsConfigPath.replace(appRootPath, '')));
   /**
-   * This special TSConfig config is all that's needed for api-extractor so it has all type information used for package:
+   * Customized TSConfig that uses `tsconfig.lib.json` as base with some required overrides:
    *
    * NOTES:
-   * - `compilerOptions.paths` doesn't work, nor is possible to turn them off when `extends` is used
+   * - `extends` is properly resolved via api-extractor which uses TS api
+   * - `skipLibCheck` needs to be explicitly set to `false` so errors propagate to api-extractor
+   * - `paths` is set to `undefined` so api-extractor won't use source files rather rollup-ed declaration files only
    *
    */
   const apiExtractorTsConfig: TsConfig = {
-    include: options.tsConfig.include,
-    /**
-     * `files` might be used to specify additional `d.ts` or global type definitions. IF they exist in package tsconfig we need to include them
-     */
-    ...(options.tsConfig.files ? { files: options.tsConfig.files } : null),
+    ...options.tsConfig,
     compilerOptions: {
+      ...options.tsConfig.compilerOptions,
```

Comment:
shouldn't options be spread at the end to override the defaults?

Reply:
no. those are "The defaults" from `tsconfig.lib.json` . we override those

Reply:
ah ok that makes sense